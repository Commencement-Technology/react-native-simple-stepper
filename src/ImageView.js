import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Image } from 'react-native';

export default class ImageView extends Component {
  static propTypes = {
    style: PropTypes.object,
    source: PropTypes.oneOfType([PropTypes.number, PropTypes.object]),
    opacity: PropTypes.number,
  };

  static defaultProps = {
    style: {},
    source: {},
    opacity: 1,
  };

  render() {
    const { style, source, opacity } = this.props;
    return <Image style={[style, { opacity }]} source={source} />;
  }
}
