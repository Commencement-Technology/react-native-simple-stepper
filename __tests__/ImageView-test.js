import React from 'react';
import ImageView from '../src/ImageView';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('ImageView', () => {
  describe('renders', () => {
    test('snapshot', () => {
      const wrapper = shallow(<ImageView />);
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
    });
  });
});
