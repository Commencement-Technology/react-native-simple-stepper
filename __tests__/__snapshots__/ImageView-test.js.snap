// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageView renders snapshot 1`] = `
<Image
  source={Object {}}
  style={
    Array [
      Object {},
      Object {
        "opacity": 1,
      },
    ]
  }
/>
`;
